File: app/src/main/java/com/example/loop/MainActivity.java
package com.example.loop;

import android.app.Activity;
import android.media.MediaPlayer;
import android.net.Uri;
import android.os.Bundle;
import android.util.Log;
import android.widget.VideoView;

public class MainActivity extends Activity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        VideoView video = new VideoView(this);
        video.setVideoURI(Uri.parse(
            "android.resource://" + getPackageName() + "/" + R.raw.z
        ));

        video.setOnPreparedListener(new MediaPlayer.OnPreparedListener() {
            @Override
            public void onPrepared(MediaPlayer mp) {
                mp.setLooping(true);
                video.start();
            }
        });

        video.setOnErrorListener(new MediaPlayer.OnErrorListener() {
            @Override
            public boolean onError(MediaPlayer mp, int what, int extra) {
                Log.e("Z-Gang", "Video error: what=" + what + " extra=" + extra);
                return true;
            }
        });

        setContentView(video);
    }
}

File: app/src/main/java/com/example/loop/MainActivity.kt
package com.example.loop

import android.app.Activity
import android.net.Uri
import android.os.Bundle
import android.widget.VideoView

class MainActivity : Activity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main)

        val videoView = findViewById<VideoView>(R.id.video)
        val videoUri = Uri.parse("android.resource://${packageName}/${R.raw.z}")

        videoView.setVideoURI(videoUri)

        videoView.setOnPreparedListener { mp ->
            mp.isLooping = true
            videoView.start()
        }

        videoView.setOnErrorListener { _, what, extra ->
            android.util.Log.e("VideoView", "Error: what=$what extra=$extra")
            true
        }
    }
}